HelpSource/Reference/JITLib_reblock.schelp

```diff
@@ -0,0 +1,213 @@
+// which they pass on to the slot objects
+Ndef(\x).put(0, { BlockSize.ir.poll(0); LFSaw.ar(123, 0, 0.3) });
+
+// you can override the Ndef settings at the slot object level
+Ndef(\x).put(0, { BlockSize.ir.poll(0); LFSaw.ar(123, 0, 0.3) }, blockSize: 32);
+
+// blockSize/upsample can be different per slot object:
+Ndef(\x).put(5, { BlockSize.ir.poll(0); LFSaw.ar(123, 0, 0.3) }, upsample: 2);
+::
+
+
+section:: Events and Patterns
+
+s.dumpOSC; // turn on dumpOSC to see the bundles sent.
+
+(blockSize: 16).play; // sends reblock new message: 66 = s_newEx.
+(upsample: 16).play;
+
+// also sends 66 s_newEx messages
+Pdef(\x, Pbind(\note, Pseq((0..7), 2), \dur, 0.125, blockSize: 31)).play;
+
+// a synthdef that produces aliasing,
+// so we hear how upsampling improves it:
+(
+SynthDef(\bsTest, { |amp=0.1, freq=(60.midicps), sustain=1|
+	var snd = LFSaw.ar(XLine.kr(freq, freq * 10, sustain, doneAction: 2), 0, amp);
+
+	BlockSize.ir.poll(0, "BlockSize");
+	(SampleRate.ir / s.sampleRate).poll(0, "upsample");
+	SampleRate.ir.poll(0, "SampleRate");
+
+	// gentle filter for upsampled sounds:
+	snd = RLPF.ar(RLPF.ar(snd, 20000, 0.7), 20000, 0.7);
+	Out.ar(0, snd);
+}).add;
+)
+
+// Events work, and this synthdef polls BlockSize etc
+(instrument: \bsTest, freq: 2000, dur: 3).play
+(instrument: \bsTest, freq: 2000, dur: 3, upsample: 4).play; // less aliasing
+(instrument: \bsTest, freq: 2000, dur: 3, upsample: 16).play; // even less
+(instrument: \bsTest, blockSize: { 16 }).play; // blockSize is evaluated
+(instrument: \bsTest, blockSize: s.options.blockSize).play
+(instrument: \bsTest, upsample: 3).play
+
+(
+PmonoArtic(
+	\default,
+	\dur, 0.15,
+	\note, Pbrown(0, 12, 3),
+	\blockSize, 16,
+	\legato, Pwhite(0.8, 1.5, 13)
+).play;
+)
+
+// also works fine in Pbinds and Pdefs
+(
+Pdef(\x,
+	Pbind(
+		\instrument, \bsTest,
+		\note, Pseq([0, 12, 24, 36], 5),
+		\dur, 1, \legato, 2,
+		\blockSize, Pseq([63, 32, 16], 7),
+		\upsample, Pseq([1, 2, 4, 8, 16], 4)
+	)
+).play
+)
+
+// playing Pdef inside Ndef works when set in Pdef...
+Ndef(\x, Pdef(\x)).play;
+
+Ndef(\y).put(4, { Saw.ar }, upsample: 3).play; // Synth
+
+
+
+/*** Which event functions get used?
+Test by blockSize check fail messages. ***/
+(blockSize: 12).play; // eventType: \note
+(blockSize: 12, dur: inf).play; // also \note
+
+(type: \grain, blockSize: 12).play; // \grain
+x = (type: \on, blockSize: 12).play; // \on
+x.release;
+
+x = (type: \play, blockSize: 12).play; // \on
+
+// \monoNote
+PmonoArtic(\default, \note, Pn(0, 1), \blockSize, 3).play;
+
+
+/******* Questions ****/
+Ndef(\z).play.put(10, \default, blockSize: 13); // Synth, but why?
+
+// how to test Event functions synthEvent:play and fadeBus?
+
+
+// When using a synthdef as source,
+// should blockSize or upsample from Ndef be used at all?
+// for clarity better not.
+Ndef(\x).blockSize = 16;
+Ndef(\x).play.put(10, \default, \blockSize, 13);
+Ndef(\x).end;
+
+// Should Pdef inherit blockSize and upsample from Ndef?
```

Comment:
I think it should.

Reply:
How best to do that? 
in the put method, do a pdef.set(\blockSize, ndef.blockSize)?

Reply:
You'd change `Event::buildForProxy` to set the respective keys in the default event. That should be all, I think.

Reply:
ok will do

Reply:
tried that, not sure if good enough - please check and improve!